### PR TITLE
[UF-443] - Header and Footer should support Errai UI

### PR DESCRIPTION
**Pull request by ederign**

### CODE REVIEW

**Comment by ederign** [comment]
I'll remove deprecated interfaces

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayoutImpl.java [hunk 1]
```diff
@@ -185,28 +141,48 @@ public HasWidgets getPerspectiveContainer() {
...
 
+    private void addFooter( IsFooter f ) {
+        if ( f instanceof FooterElement ) {
+            footerPanel.add( extractWidgetFrom( ( IsElement ) f ) );
+        } else if ( f instanceof Footer ) {
+            footerPanel.add( ( IsWidget ) f );
+        }
+    }
+
+    ElementWrapperWidget<?> extractWidgetFrom( IsElement h ) {
```
Comment (csadilek) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayoutImpl.java]: Better name would be createWidgetFrom/For or wrapElement...as nothing here gets extracted.

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/IsHeader.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.uberfire.client.workbench;
+
+/**
+ * Parent Type of all Headers implementations
+ */
+public interface IsHeader extends Orderable{
```
Comment (csadilek) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/IsHeader.java]: Orderable {

**Comment by ederign** [comment]
JENKINS RETEST THIS

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayoutImpl.java [hunk 3]
```diff
@@ -141,36 +83,55 @@ public int getClientWidth() {
      * it's cleared and repopulated with the new perspective's root view each time
      * {@link #setHeaderContents(java.util.List)} gets called.
      */
-    private final Panel headerPanel = new FlowPanel();
+    private Div headerPanel;
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayoutImpl.java]: @csadilek that is new from your last review.

**Comment by ederign** [comment]
*Do not merge*. I still need to migrate all header implementations in the workbench.

**Comment by ederign** [comment]
@csadilek related PR's

https://github.com/dashbuilder/dashbuilder/pull/280
https://github.com/droolsjbpm/kie-wb-common/pull/649
https://github.com/droolsjbpm/jbpm-form-modeler/pull/85
https://github.com/droolsjbpm/drools-wb/pull/392
https://github.com/droolsjbpm/optaplanner-wb/pull/133
https://github.com/droolsjbpm/jbpm-designer/pull/542
https://github.com/droolsjbpm/jbpm-console-ng/pull/615
https://github.com/droolsjbpm/appformer/pull/69
https://github.com/droolsjbpm/kie-wb-distributions/pull/452

**Pull request merged**